spec/api_spec.rb

```diff
@@ -57,4 +57,85 @@
     PayEx::API.parse_params({ a: 1 }, { 'b' => { default: 2 } }).
       should == { 'a' => 1, 'b' => 2 }
   end
+
+  describe '#parse_param' do
+    context 'when format validation fails' do
+      context 'when resolve proc is present' do
+        it 'calls the proc' do
```

Comment:
Why shoulda matchers?

Provided you can I would probably write something like:
```ruby
resolver = -> (v) { v[0..2].strip }
options = { format: /^[a-z]{,3}$/, resolve: resolver }
allow(resolver).to receive(:call).and_return('too')

PayEx::API.parse_param('too long param', options)

expect(resolver).to have_received(:call).with('too long param')
```

Reply:
It's because we're running rspec 2.11.

Reply:
Gotcha. I suspected something like that. Feel free to merge :)